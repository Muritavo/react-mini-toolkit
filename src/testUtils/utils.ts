import path from "path";
import * as ts from "typescript";

function workaroundForPathBecauseIDnotKnowTSCompilersYet(path: string) {
    if (path.startsWith("/node_modules"))
        return `.${path}`;
    else
        return path;
}

export function getDiagnosticsForText(files: {
    [filename: string]: string
}) {
    const fileNames = Object.keys(files);
    const asts = fileNames.reduce((r, filename) => ({
        ...r,
        [filename]: ts.createSourceFile(filename, files[filename], ts.ScriptTarget.ESNext)
    }), {}) as { [key: string]: ts.SourceFile }
    const options: ts.CompilerOptions = {
        target: ts.ScriptTarget.ESNext,
        module: ts.ModuleKind.CommonJS,
        moduleResolution: ts.ModuleResolutionKind.NodeJs,
        jsx: ts.JsxEmit.ReactJSX,
        esModuleInterop: true,
    };
    const nativeHost = ts.createCompilerHost(options);
    function fileIsMocked(filePath: string) {
        return fileNames.includes(filePath)
    }
    function cleanFilePath(f: string) {
        return f.replace(/^[^A-Za-z]/g, "");
    }
    const host: ts.CompilerHost = {
        fileExists: filePath => fileIsMocked(cleanFilePath(filePath)) || nativeHost.fileExists(workaroundForPathBecauseIDnotKnowTSCompilersYet(filePath)),
        directoryExists: dirPath => dirPath === "/" || nativeHost.directoryExists!(workaroundForPathBecauseIDnotKnowTSCompilersYet(dirPath)),
        getCurrentDirectory: () => "/",
        getDirectories: (s) => nativeHost.getDirectories!(s),
        getCanonicalFileName: fileName => fileName,
        getNewLine: () => "\n",
        getDefaultLibFileName: () => nativeHost.getDefaultLibFileName(options),
        getSourceFile: filePath => fileIsMocked(cleanFilePath(filePath)) ? asts[cleanFilePath(filePath)] : nativeHost.getSourceFile(workaroundForPathBecauseIDnotKnowTSCompilersYet(filePath), ts.ScriptTarget.Latest),
        readFile: filePath => fileIsMocked(cleanFilePath(filePath)) ? files[cleanFilePath(filePath)] : nativeHost.readFile(workaroundForPathBecauseIDnotKnowTSCompilersYet(filePath)),
        useCaseSensitiveFileNames: () => true,
        writeFile: () => { }
    };
    const program = ts.createProgram({
        options,
        rootNames: fileNames.concat(require.resolve("@types/jest/index.d.ts")),
        host
    });

    return ts.getPreEmitDiagnostics(program);
}