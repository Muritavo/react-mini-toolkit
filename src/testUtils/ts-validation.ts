import { getDiagnosticsForText } from "../testUtils/utils";

export function assertFilesCompile(outputFiles: {
    [fileName: string]: string
}) {
  const outputResult = getDiagnosticsForText(outputFiles).map(
    (o) => o.messageText
  );

  const errors = outputResult.filter((o) => {
    const a = JSON.stringify(o);
    return (
      !a.includes(`MockComponent.module.scss'`) &&
      !a.includes("README.md") &&
      !a.includes("'cypress'") &&
      !a.includes("'Cypress'") &&
      !a.includes("'JQuery'")
    );
  });
  expect(errors).toHaveLength(0);
}
