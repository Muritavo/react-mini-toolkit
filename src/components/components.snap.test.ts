import { getDiagnosticsForText } from "../testUtils/utils";
import { FEATURES, generateOutput } from ".";

const FEATURES_TO_TEST = FEATURES.map((f) => f.name);

const boolCombo = (size: number) => {
  const buf = Array(1 << size);
  for (let i = buf.length; i--; ) {
    buf[i] = Array(size);
    for (let j = size; j--; ) buf[i][j] = !!+!!(i & (1 << j));
  }
  return buf;
};

const allCombinations = boolCombo(FEATURES_TO_TEST.length);
const allFeatureCombinations = allCombinations.map((combination) =>
  FEATURES_TO_TEST.reduce((r, key, i) => ({ ...r, [key]: combination[i] }), {})
) as FeatureMap[];

function buildMockedFeatureData(featureMap: FeatureMap) {
  const map: FeatureData = {};
  if (featureMap.documentation)
    map.documentation = {
      businessRules: ["SOME MOCKED BUSINESS RULE", "ANOTHER ONE"],
    };
  return map;
}

it.each(allFeatureCombinations)(
  "Features for the component generation when features are enabled as %s",
  (enabledFeatures) => {
    const outputFiles = generateOutput(
      "MockComponent",
      "A component to check that everything is fine",
      enabledFeatures as any,
      buildMockedFeatureData(enabledFeatures)
    );
    expect(outputFiles).toMatchSnapshot();

    const outputResult = getDiagnosticsForText(outputFiles).map(
      (o) => o.messageText
    );
    const errors = outputResult.filter((o) => {
      const a = JSON.stringify(o);
      !a.includes(`MockComponent.module.scss'`) && !a.includes("README.md");
    });
    expect(errors).toHaveLength(0);
  }
);
