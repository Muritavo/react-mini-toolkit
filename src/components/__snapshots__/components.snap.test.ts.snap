// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features for the component generation when features are enabled as { tests: false, scss: false, storybook: false, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: false, storybook: false, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: false, storybook: true, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: false, storybook: true, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: true, storybook: false, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: true, storybook: false, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: true, storybook: true, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: false, scss: true, storybook: true, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: false, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';

import Component from './MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: false, intl: false } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';

import Component from 'folder/MockComponent/MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: false, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import Component from './MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: false, intl: true } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';
import { IntlProvider } from 'react-intl';
import Component from 'folder/MockComponent/MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: true, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';

import { InitialImplementation as Component } from './MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: true, intl: false } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';

import { InitialImplementation as Component } from 'folder/MockComponent/MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';



/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: true, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { InitialImplementation as Component } from './MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: false, storybook: true, intl: true } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';
import { IntlProvider } from 'react-intl';
import { InitialImplementation as Component } from 'folder/MockComponent/MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';

import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: false, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';

import Component from './MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: false, intl: false } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';

import Component from 'folder/MockComponent/MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: false, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import Component from './MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: false, intl: true } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';
import { IntlProvider } from 'react-intl';
import Component from 'folder/MockComponent/MockComponent';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: true, intl: false } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';

import { InitialImplementation as Component } from './MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: true, intl: false } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';

import { InitialImplementation as Component } from 'folder/MockComponent/MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<Component {...props}/>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';


/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: true, intl: true } 1`] = `
Object {
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import { InitialImplementation as Component } from './MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return render(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it.todo(\\"SOME MOCKED BUSINESS RULE\\");
it.todo(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;

exports[`Features for the component generation when features are enabled as { tests: true, scss: true, storybook: true, intl: true } 2`] = `
Object {
  "/root/path/to/project/cypress/integration/folder/MockComponent/MockComponent.test.tsx": "import React from 'react';
import { mount } from 'cypress/react';
import { IntlProvider } from 'react-intl';
import { InitialImplementation as Component } from 'folder/MockComponent/MockComponent.stories';

function renderScreen(props: React.ComponentProps<typeof Component>) {
  return mount(<IntlProvider locale=\\"pt-br\\">
            <Component {...props}/>
        </IntlProvider>);
}

it('Should at least render :)', () => {
    renderScreen({});
})

it(\\"SOME MOCKED BUSINESS RULE\\");
it(\\"ANOTHER ONE\\");",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.module.scss": "",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.stories.tsx": "import React from 'react';
import MockComponent from \\"./MockComponent\\";
    
export default {
    component: MockComponent,
    title: \\"Mock Component\\"
}

export const InitialImplementation = (args: any) => <MockComponent {...args}/>;
InitialImplementation.args = {} as Partial<React.ComponentProps<typeof MockComponent>>",
  "/root/path/to/project/src/folder/MockComponent/MockComponent.tsx": "import React from 'react';
import Styles from './MockComponent.module.scss';
import { useIntl } from 'react-intl'
import messages from './messages'

/**
 * A component to check that everything is fine
 **/
export default function MockComponent() {
    const { formatMessage } = useIntl();
    return <>MockComponent</>;
}",
  "/root/path/to/project/src/folder/MockComponent/index.tsx": "export { default } from './MockComponent';
",
  "/root/path/to/project/src/folder/MockComponent/messages.ts": "import { defineMessages } from 'react-intl';

export default defineMessages({})",
}
`;
